EpiAware/docs/src/showcase/replications/chatzilena-2019/index.jl

```diff
@@ -0,0 +1,622 @@
+```
+Where $\mu_t = \log(I(t))$.
+
+We modify this approach since it implies that the $\mu_t$ is treated as constant between observation times.
+
+Instead we redefine $\kappa_t$ as the log-residual:
+
+$\kappa_t = \log(\lambda_t / I(t)).$
+
+With the transition density:
+
+```math
+\kappa_{t+1} | \kappa_t \sim N\Big(\kappa_te^{-\phi}, {\sigma^2 \over 2 \phi} \left(1 - e^{-2\phi} \right)\Big).
+```
+
+This is an AR(1) process.
+
+The stochastic model is completed:
+
+```math
+\begin{aligned}
+Y_t &\sim \text{Poisson}(\lambda_t)\\
+\lambda_t &= I(t)\exp(\kappa_t)\\
+\beta &\sim \text{LogNormal}(\text{logmean}=0,\text{logstd}=1) \\
+\gamma & \sim \text{Gamma}(\text{shape} = 0.004, \text{scale} = 50)\\
+S(0) /N &\sim \text{Beta}(0.5, 0.5)\\
+\phi & \sim \text{HalfNormal}(0, 100) \\
+1 / \sigma^2 & \sim \text{InvGamma}(0.1,0.1).
+\end{aligned}
+```
+
+"
+
+# ╔═╡ 69ba59d1-2221-463f-8853-ae172739e512
+md"
+We will using the `AR` struct from `EpiAware` to define the auto-regressive process in this model which has a direct parameterisation of the `AR` model.
+
+To convert from the formulation above we sample from the priors, and define `HalfNormal` priors based on the sampled prior means of $e^{-\phi}$ and ${\sigma^2 \over 2 \phi} \left(1 - e^{-2\phi} \right)$. We also add a strong prior that $\kappa_1 \approx 0$.
+"
+
+# ╔═╡ 178e0048-069a-4953-bb24-5116eb81cc41
+ϕs = rand(truncated(Normal(0, 100), lower = 0.0), 1000)
+
+# ╔═╡ e6bcf0c0-3cc4-41f3-ad20-fa11bf2ca37b
+σ²s = rand(InverseGamma(0.1, 0.1), 1000) .|> x -> 1 / x
+
+# ╔═╡ 4f07e8ba-30d0-411f-8c3e-b6d5bc1bb5fa
+sampled_AR_damps = ϕs .|> ϕ -> exp(-ϕ)
+
+# ╔═╡ 48032d21-53fa-4c0a-85cb-c22327b55073
+sampled_AR_stds = map(ϕs, σ²s) do ϕ, σ²
+    (1 - exp(-2 * ϕ)) * σ² / (2 * ϕ)
+end
+
+# ╔═╡ 89c767b8-97a0-45bb-9e9f-821879ddd38b
+md"
+We define the AR(1) process by matching means of `HalfNormal` prior distributions for the damp parameters and std deviation parameter to the calculated the prior means from the _Chatzilena et al_ definition.
+"
+
+# ╔═╡ 71a26408-1c26-46cf-bc72-c6ba528dfadd
+ar = AR(
+    damp_priors = [HalfNormal(mean(sampled_AR_damps))],
+    std_prior = HalfNormal(mean(sampled_AR_stds)),
+    init_priors = [Normal(0, 0.001)]
+)
+
+# ╔═╡ 9309f7f8-0896-4686-8bfc-b9f82d91bc0f
+@model function stochastic_ode_mdl(Yt, ts, logobsprob, obs, prob, N;
+        solver = AutoTsit5(Rosenbrock23()),
+        upjitter = 1e-2#0.1,
+)
+
+    ##Priors##
+    β ~ LogNormal(0.0, 1.0)
+    γ ~ Gamma(0.004, 1 / 0.002)
+    S₀ ~ Beta(0.5, 0.5)
+
+    ##Remake ODE model##
+    _prob = remake(prob;
+        u0 = [S₀, 1 - S₀, 0.0],
+        p = [β, γ]
+    )
+
+    ##Solve ODE model##
+    sol = solve(_prob, solver;
+        saveat = ts,
+        verbose = false
+    )
+
+    ##Sample the log-residual AR process##
+    nobs = length(Yt)
+    @submodel κₜ = generate_latent(logobsprob, nobs)
+    λt = @. N * sol[2, :] * exp(κₜ) + upjitter
```

Comment:
Can we be pro and use any of the helpers (transformLatent eetc etc) if no can we have an issue to make one that would let us do this?

Reply:
Yes, really good shout